src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import './Modal.css';
+import { ModalProps, ModalState } from '../../types';
+
+class Modal extends React.Component<ModalProps, ModalState> {
+  private modalRef: React.RefObject<HTMLDialogElement>;
+
+  constructor(props: ModalProps) {
+    super(props);
+    this.modalRef = React.createRef<HTMLDialogElement>();
+  }
+
+  componentDidMount() {
+    if (this.modalRef.current) {
+      this.modalRef.current.showModal();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.modalRef.current) {
+      this.modalRef.current.showModal();
+    }
+  }
+
+  closeModal() {
+    if (this.modalRef.current) {
+      this.modalRef.current.close();
```

Comment:
```suggestion
      this.modalRef.current?.close();
```
옵셔널체이닝으로 해볼 수도 있을 것 같아요